Examples/LambdaFunctions/Sources/LambdaAPI/ProductLambdaHandler.swift

```diff
@@ -0,0 +1,222 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftAWSLambdaRuntime open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftAWSLambdaRuntime project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftAWSLambdaRuntime project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import Foundation
+#if canImport(FoundationNetworking)
+import FoundationNetworking
+#endif
+import AWSLambdaRuntime
+import NIO
+import Logging
+import AWSLambdaEvents
+
+struct ProductLambdaHandler: EventLoopLambdaHandler {
+    
+    typealias In = APIGateway.V2.Request
+    typealias Out = APIGateway.V2.Response
+    
+    let service: ProductService
+    let operation: Operation
+    
+    func handle(context: Lambda.Context, event: APIGateway.V2.Request) -> EventLoopFuture<APIGateway.V2.Response> {
+    
+        switch operation {
+        case .create:
+            logger.info("create")
+            let create = CreateLambdaHandler(service: service).handle(context: context, event: event)
+                .flatMap { response -> EventLoopFuture<APIGateway.V2.Response> in
+                switch response {
+                case .success(let result):
+                    let value = APIGateway.V2.Response(with: result, statusCode: .created)
+                    return context.eventLoop.makeSucceededFuture(value)
+                case .failure(let error):
+                    let value = APIGateway.V2.Response(with: error, statusCode: .forbidden)
```

Comment:
is the error necessarily permission related? or would we refactor the error handling to more precisely set the status code based on the error?

Reply:
Refactored in 8649522
I need to investigate more if we can map the status code with the error